framework/CodeInterfaces/PHISICS/phisicsdata.py

```diff
@@ -0,0 +1,961 @@
+    """
+    timeSteps = []
+    with open(os.path.join(self.workingDir,self.mrtauOutputFileMPI[0]), 'r') as outfile:
+      for line in outfile:
+        stringIsFloatNumber = self.isFloatNumber(line.split(','))
+        if stringIsFloatNumber is True:
+          timeSteps.append(line.split(',')[0])
+    return timeSteps
+    
+  def getMrtauIsotopeList(self,mrtauCSVOutput):
+    """
+      Gets the isotope in the MRTAU standalone output.
+      @ In, mrtauCSVOutput, string, MRTAU file name
+      @ Out, None
+    """
+    self.isotopeListMrtau = []
+    with open(os.path.join(self.workingDir,self.mrtauOutputFileMPI[0]), 'r') as outfile:
+      for line in outfile:
+        if re.search(r'TIME',line):
+          line = re.sub(r'TIME\s+\(days\)',r'',line)
+          line = re.sub(r' ',r'',line)
+          line = re.sub(r'\n',r'',line)
+          self.isotopeListMrtau = filter(None,line.split(','))
+          break
+    
+  def getInstantTimeSteps(self,input):
+    """
+      Gets the time steps in the INSTANT output.
+      @ In, input, string, input file name
+      @ Out, timeSteps, list, list of time steps
+    """
+    count = 0
+    timeSteps = []
+    with open(os.path.join(self.workingDir,input), 'r') as outfile:
+      for line in outfile :
+        if re.search(r'PHISICS',line):
+          timeSteps.append(line.split(',')[line.split(',').index('Time') + 1])
+    return timeSteps
+
+  def reduceDigits(self,mrtauTimeSteps):
+    """
+      Reduces the number of digits in the MRTAU time steps.
+      @ In, mrtauTimeSteps, list
+      @ Out, timeSteps, list, list of time steps
+    """
+    timeSteps = []
+    for i in xrange (0, len(mrtauTimeSteps)):
+      timeSteps.append('%.6E' % (float(mrtauTimeSteps[i])))
+    return timeSteps
+  
+  def findUnits(self,line):
+    """
+      Returns the units in which the time steps are printed in the INSTANT output
+      @ In, line, string, line containing the units
+      @ Out, units, string
+    """ 
+    units = line.split(' ')[-3]
+    return units
+  
+  def getKeff (self):
+    """
+      Gets the multiplication factor.
+      @ In, None
+      @ Out, keff, string, floating number under string format of keff
+      @ Out, errorKeff, string, floating number under string format of errorKeff
+    """
+    self.units = None
+    keff = []
+    with open(os.path.join(self.workingDir,self.instantOutputFileMPI[0]), 'r') as outfile:
+      for line in outfile:
+        if re.search(r'Time Input Type', line):
+          self.units = self.findUnits(line)
+        if re.search(r'k-effective at the last', line) :
+          keff = [line.split()[-1]]
+        if re.search(r'error for the eigenvalue', line) :
+            errorKeff = [line.split()[-1]]
+    return keff, errorKeff
+
+  def isNumber(self,line):
+    """
+      Checks if a string is an integer.
+      @ In, line, list, list of strings
+      @ Out, isNumber, Boolean, True if integer, False otherwise
+    """
+    if line != []:
```

Comment:
```isinstance```

PS. if is it a list????

Reply:
if the list is empty it return an error. So I excluded the empty list case (happens if a blank line is considered)